### ISSUE TITLE
Booking is failing with product exception

### ISSUE THREAD

**Issue by pmorelli92** [hotelbeds-sdk/hotel-api-sdk-Net#43]
Hi guys, 

I am having an error while doing bookings, let me attach the request:

```c#
{
	"holder": {
		"name": "TEST",
		"surname": "TEST"
	},
	"rooms": [{
		"paxes": [{
			"roomId": 1,
			"type": "AD",
			"age": 46,
			"name": "TEST",
			"surname": "TEST"
		},
		{
			"roomId": 1,
			"type": "AD",
			"age": 30,
			"name": "Unknown",
			"surname": "Unknown"
		}],
		"rateKey": "20161202|20161209|W|197|18968|DBL.ST|OFFER EB ST|BB||1~2~0||N@7F2F821B2716490F9E849B9610E69DC4"
	}],
	"paymentData": null,
	"clientReference": "OR-1480106168646-1",
	"remark": "TEST REMARKS",
	"echoToken": null,
	"language": null
}
```

The response I am getting is the following:

> ProductException(ae31f13f-8a20-4979-a82b-1980663ab6fc, AVAILABILITY_CONFIRM, There is not enough space in the service for all the passengers.):

This is on the testing environment.
Is this a problem of the SDK?

Thanks,
Regards

**Comment by alvmunoz** [comment]
Hi @pmorelli92 , we will look into it.

**Comment by alvmunoz** [comment]
Can you provide me your apikey ?  (not the secret)

**Comment by pmorelli92** [comment]
Hi @alvmunoz my ApiKey is the following: **wbxrv2d759hmvh4b4wu6jxyu**

**Comment by alvmunoz** [comment]
I cannot this rateKey available. Does it happen to you only with this specific case? Or every time? 

Normally you get this error when you tried to confirm something which allotment was finished right before you book it.  
That is, there is availability but it gets booked by someone else before you do it. 

I cannot see it available now, which is correct according to the message you received in confirmation.  I guess you cannot see it too. 

Do you find this issue with other rateKeys often?  Availablity in test environment is not updated as fast as in LIVE environment.  Which means that we might sold out the allotment in a specific hotel but still display it in availability for  some minutes before it expires.

**Comment by pmorelli92** [comment]
Hi @alvmunoz today I was trying to reproduce the error but at first I was having another error.

> Unable to translate bytes [E9] at index 827 from specified code page to Unicode.

I was getting this error while trying to do a booking to the hotel: 18968
Moreover, if I check the booking list, those bookings were confirmed but I got an exception instead of the bookingRS message from the SDK.

Then after two or three attempts I started getting the "ProductException" error (I guess the hotel run out of allotment but still returns the availability as you stated)

Can we do something about the new error?

> Unable to translate bytes [E9] at index 827 from specified code page to Unicode.

[Some error images](http://imgur.com/a/ZaTrd)

**Comment by pmorelli92** [comment]
@alvmunoz I was able to isolate the error.

The booking request is:

```json
{
  "holder": {
    "name": "Rosetta",
    "surname": "Pruebas"
  },
  "rooms": [
    {
      "paxes": [
        {
          "roomId": 1,
          "type": "AD",
          "age": 30,
          "name": "NombrePasajero1",
          "surname": "ApellidoPasajero1"
        },
        {
          "roomId": 1,
          "type": "AD",
          "age": 30,
          "name": "NombrePasajero2",
          "surname": "ApellidoPasajero2"
        }
      ],
      "rateKey": "20170207|20170214|W|197|18968|DBL.ST|OFFER EB ST|BB||1~2~0||N@B979B0560D484F1C9D0375A2E983804B"
    }
  ],
  "clientReference": "SDK Test",
  "remark": "***SDK***TESTING"
}
```

And the response is the following:

```json
{
	"auditData": {
		"processTime": "2082",
		"timestamp": "2016-12-02 19:15:08.031",
		"requestHost": "181.164.93.206",
		"serverId": "sa37AUX3ROLBLIS.env",
		"environment": "[int]",
		"release": "8629cf6ef0a2deed178b86daa0e13c79b5c99237",
		"internal": "0|B979B0560D484F1C9D0375A2E983804B|MX||1|1|||||||||||R|1|7||||||wbxrv2d759hmvh4b4wu6jxyu||"
	},
	"booking": {
		"reference": "197-2985570",
		"clientReference": "SDK TEST",
		"creationDate": "2016-12-02",
		"status": "CONFIRMED",
		"modificationPolicies": {
			"cancellation": true,
			"modification": true
		},
		"creationUser": "wbxrv2d759hmvh4b4wu6jxyu",
		"holder": {
			"name": "ROSETTA",
			"surname": "PRUEBAS"
		},
		"hotel": {
			"checkIn": "2017-02-07",
			"checkOut": "2017-02-14",
			"code": 18968,
			"name": "Elysees Union",
			"categoryCode": "3EST",
			"categoryName": "3 STARS",
			"destinationCode": "PAR",
			"destinationName": "Paris",
			"zoneCode": 14,
			"zoneName": "Arr8/16:Etoile-C.Elys�es/Trocadero",
			"latitude": "48.868387",
			"longitude": "2.291788",
			"rooms": [{
					"status": "CONFIRMED",
					"id": 1,
					"code": "DBL.ST",
					"name": "DOUBLE STANDARD",
					"paxes": [{
							"roomId": 1,
							"type": "AD",
							"name": "NombrePasajero2",
							"surname": "ApellidoPasajero2"
						}, {
							"roomId": 1,
							"type": "AD",
							"name": "NombrePasajero1",
							"surname": "ApellidoPasajero1"
						}
					],
					"rates": [{
							"rateKey": "20170207|20170214|W|197|18968|DBL.ST|OFFER EB ST|BB||1~2~0|30~30|N@0",
							"rateClass": "NRF",
							"net": "771.01",
							"rateComments": "1x DOUBLE Estimated total amount of taxes & fees for this booking: 23.10 Euro   payable on arrival  . NO AMENDMENT OR CANCELLATION WILL BE ACCEPTED ONCE BOOKINGS ARE MADE UNDER THIS NONREFUNDABLE PROMOTION.",
							"paymentType": "AT_WEB",
							"packaging": false,
							"boardCode": "BB",
							"boardName": "BED AND BREAKFAST",
							"cancellationPolicies": [{
									"amount": "771.01",
									"from": "2016-12-01T23:59:00+01:00"
								}
							],
							"rateBreakDown": {
								"rateDiscounts": [{
										"code": "SD",
										"name": "SPECIAL DISCOUNT",
										"amount": "-85.69"
									}
								]
							},
							"rooms": 1,
							"childrenAges": ""
						}
					]
				}
			],
			"totalNet": "771.01",
			"currency": "USD",
			"supplier": {
				"name": "HOTELBEDS PRODUCT,S.L.U.",
				"vatNumber": "ESB38877676"
			}
		},
		"remark": "***SDK***TESTING",
		"totalNet": 771.01,
		"pendingAmount": 771.01,
		"currency": "USD"
	}
}
```

As you can see in the zone name we found:

> "zoneName":  "Arr8/16:Etoile-C.Elys�es/Trocadero",

And this is leading to the folowing error:

> Unable to translate bytes [E9] at index 827 from specified code page to Unicode.

**Comment by pmorelli92** [comment]
Added the fix in the pull request: #45

@alvmunoz  Can we see this change added into a new package?

Regards,

**Comment by alvmunoz** [comment]
That's actually a bug in the API, but I'll accept your change as a workaround until this is fixed :+1:

**Comment by pmorelli92** [comment]
Hi @alvmunoz can we get this change in a package?

**Comment by alvmunoz** [comment]
For sure, I will try to have it within today.

**Comment by pmorelli92** [comment]
Hi @alvmunoz, sorry for being insistent, but our client is putting pressure on this issue. Can we get this fix deployed? Thanks for the understanding.

**Comment by alvmunoz** [comment]
@pmorelli92 ,
sorry for the delay, just published version 0.6.6 with your fix.
Thank you very much.

**Comment by pmorelli92** [comment]
Thank you!
